pkgs/development/libraries/science/math/magma/generic.nix

```diff
@@ -129,30 +145,53 @@ stdenv.mkDerivation {
   ];
 
   cmakeFlags = [
-    "-DGPU_TARGET=${gpuTargetString}"
-    (lib.cmakeBool "MAGMA_ENABLE_CUDA" cudaSupport)
-    (lib.cmakeBool "MAGMA_ENABLE_HIP" rocmSupport)
-  ] ++ lists.optionals static [
-    "-DBUILD_SHARED_LIBS=OFF"
+    (strings.cmakeFeature "GPU_TARGET" gpuTargetString)
```

Comment:
why not

```suggestion
    (lib.cmakeFeature "GPU_TARGET" gpuTargetString)
```

?

Reply:
Cons of referencing `strings`: this way `strings` can't be private anymore, `lib` has to have `strings`; we can't move `cmakeBool` out of `strings`
Cons of referencing `lib.${name}`: we postulate the existence of the fat/throw all-in `lib`, we can't generally call the recipes with a smaller `lib`

Consistency-wise, my impression is that it's more common in Nixpkgs to use `lib`; inside `cuda-modules/`, it's more frequent to see `lists.${name}`, `strings.${name}` since recently; I personally don't care much

Reply:
Purely preference -- I despise `lib` being a catch-all which re-exports everything available in its constituent attribute sets.

Reply:
Yet another bikeshed. Bye!